heft-plugins/heft-lint-plugin/src/SarifFormatter.ts

```diff
@@ -0,0 +1,258 @@
+  tool: {
+    driver: {
+      name: 'ESLint',
+      informationUri: 'https://eslint.org',
+      rules: []
+    }
+  }
+};
+
+const sarifLog: ISarifLog = {
+  version: '2.1.0',
+  $schema: 'http://json.schemastore.org/sarif-2.1.0-rtm.5',
+  runs: [sarifRun]
+};
+
+// Main function
+export function formatAsSARIF(results: IExtendedLintResult[], options: ISerifFormatterOptions): ISarifLog {
+  const { ignoreSuppressed, eslintVersion } = options;
+
+  if (typeof eslintVersion !== undefined) {
+    sarifRun.tool.driver.version = eslintVersion;
+  }
+  let executionSuccessful: boolean = true;
+
+  for (const result of results) {
+    const { filePath } = result;
+    let sarifFile: ISarifFile | undefined = sarifFiles.get(filePath);
+    if (sarifFile === undefined) {
+      const artifactIndex: number = sarifFiles.size;
+      const fileUrl: string = url.pathToFileURL(filePath).toString();
+
+      sarifFile = {
+        location: {
+          uri: fileUrl
+        }
+      };
+      sarifFiles.set(filePath, sarifFile);
+
+      const containsSuppressedMessages: boolean =
+        result.suppressedMessages && result.suppressedMessages.length > 0;
+      const messages: IMessage[] =
+        containsSuppressedMessages && !ignoreSuppressed
+          ? [...result.messages, ...result.suppressedMessages]
+          : result.messages;
+
+      if (messages.length > 0) {
+        for (const message of messages) {
+          const level: string = message.fatal || message.severity === 2 ? 'error' : 'warning';
+          const sarifRepresentation: ISarifRepresentation = {
+            level,
+            message: {
+              text: message.message
+            },
+            locations: [
+              {
+                physicalLocation: {
+                  artifactLocation: {
+                    uri: fileUrl,
+                    index: artifactIndex
+                  }
+                }
+              }
+            ]
+          };
+
+          if (message.ruleId) {
+            sarifRepresentation.ruleId = message.ruleId;
+
+            if (containsSuppressedMessages && !ignoreSuppressed) {
+              sarifRepresentation.suppressions = message.suppressions
+                ? message.suppressions.map((suppression: ISuppressedAnalysis) => {
+                    return {
+                      kind: suppression.kind === 'directive' ? 'inSource' : 'external',
+                      justification: suppression.justification
+                    };
+                  })
+                : [];
+            }
+          } else {
+            sarifRepresentation.descriptor = {
+              id: internalErrorId
+            };
+
+            if (sarifRepresentation.level === 'error') {
+              executionSuccessful = false;
+            }
+          }
+
+          const physicalLocation: ISarifPhysicalLocation = sarifRepresentation.locations[0].physicalLocation;
+
+          if (message.line! > 0 || message.column! > 0) {
```

Comment:
I think you can get away with `message.line || message.column` here since 0 is falsy

Reply:
Same goes for the checks below, ex `startLine: line || undefined`